parser/src/lib.rs

```diff
@@ -517,7 +517,7 @@ impl<'input, I, Id, F> ParserEnv<I, F>
                                   sep_end_by(self.expr(), token(Token::Comma)))
                          .map(|exprs| {
                              loc(Expr::Array(Array {
-                                 id: self.empty_id.clone(),
+                                 typ: Type::hole(),
```

Comment:
Haven't figured out how to cache these yet. I needed to add a `hole_typ: AstType<<F::Ident as AstId>::Untyped>` field to `ParserEnv`, but `F::Ident` is not known to be an `AstId`. Any suggestions?